tests/core/full_node/test_mempool_performance.py

```diff
@@ -69,12 +92,24 @@ async def test_mempool_update_performance(self, bt, wallet_nodes_mempool_perf, d
         blocks = bt.get_consecutive_blocks(3, blocks)
         await full_node_api_1.full_node.respond_block(full_node_protocol.RespondBlock(blocks[-3]))
 
+        fast_durations: List[float] = []
+        slow_durations: List[float] = []
+
         for idx, block in enumerate(blocks):
             start_t_2 = time.time()
             await full_node_api_1.full_node.respond_block(full_node_protocol.RespondBlock(block))
             end_t_2 = time.time()
             duration = end_t_2 - start_t_2
             if idx >= len(blocks) - 3:
-                assert duration < 0.1
+                slow_durations.append(duration)
             else:
-                assert duration < 0.0003
+                fast_durations.append(duration)
+
+        fast_statistics = Statistics.create(data=fast_durations)
+        slow_statistics = Statistics.create(data=slow_durations)
+
+        message = f"fast: {fast_statistics}\n\nslow: {slow_statistics}"
+        print(f"\n\n{message}")
+        assert (
+            fast_statistics.percentile_90 < 0.00005 and slow_statistics.percentile_90 < 0.1
```

Comment:
90th percentile on 3 samples isn't clearly a good idea.  Maybe it's ok though since it seems to keep 2 and ignore 1?